src/common/module_builder.cc

```diff
@@ -135,7 +145,46 @@ void ModuleBuilder::convertFromVHLOToSHLO(
   }
 
   DLOG_F(LOG_DEBUG, "SHLO Module:");
-  print_module(mlir_module);
+  printModule(mlir_module);
+}
+
+void ModuleBuilder::collectOutputTypes(
+    const mlir::OwningOpRef<mlir::ModuleOp> &module) {
+  DLOG_F(LOG_DEBUG, "ModuleBuilder::collectOutputTypes");
+
+  m_is_output_scalar.clear();
+
+  module.get().walk([&](mlir::Operation *op) {
+    mlir::func::FuncOp funcOp = mlir::dyn_cast<mlir::func::FuncOp>(op);
+    mlir::ModuleOp moduleOp = mlir::dyn_cast<mlir::ModuleOp>(op);
+
+    // We care only about return ops of public functions, as that are the ones
+    // that will produce results in the flatbuffer.
+    if (funcOp && funcOp.isPublic()) {
```

Comment:
Change to `if (!condition) return`, see more: https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code